sdk/core/azure-core-amqp/test/ut/amqp_value_tests.cpp

```diff
@@ -188,6 +245,7 @@ TEST_F(TestValues, TestChar)
   {
     AmqpValue value{AmqpValue::CreateChar(37)};
```

Comment:
Should we also write a few test cases with different unicode char literals, at least for illustrative puropses?
https://stackoverflow.com/questions/6796157/unicode-encoding-for-string-literals-in-c11

For examples, "\u2764\uFE0F" is the same thing as "\xE2\x9D\xA4\xEF\xB8\x8F", they mean the same character (heart emoji), and it probably can fit into char32_t.

Reply:
I believe that "\u2764\uFE0F" is not a UTF-32 encoded value, I believe it's 2 UTF-16 codepoints. The UTF-32 encoded value for the heart emoji is [0x1f48c](https://unicode.org/emoji/charts/full-emoji-list.html#1f48c)., so it is the equivalent of calling `CreateChar(0x1f48c)`. I'm not sure I see a semantic difference between `CreateChar(37)` and `CreateChar(0x1f48c)`.

The underlying AMQP "Char" type is defined as a 32 bit unsigned integer value. There are no additional semantics defined for it (and most importantly, there are no conversion semantics defined for AMQP Char values, so it doesn't make sense to add tests for it).